lib/PhasarLLVM/TaintConfig/LLVMTaintConfig.cpp

```diff
@@ -72,127 +72,106 @@ void LLVMTaintConfig::addAllFunctions(const LLVMProjectIRDB &IRDB,
 
     const auto *Fun = FnDefs[0];
 
-    // handle a function's parameters
-    if (FunDesc.contains("params")) {
-      auto Params = FunDesc["params"];
-      if (Params.contains("source")) {
-        for (unsigned Idx : Params["source"]) {
-          if (Idx >= Fun->arg_size()) {
-            llvm::errs()
-                << "ERROR: The source-function parameter index is out of "
-                   "bounds: "
-                << Idx << "\n";
-            // Use 'continue' instead of 'break' to get error messages for the
-            // remaining parameters as well
-            continue;
-          }
-          addTaintCategory(Fun->getArg(Idx), TaintCategory::Source);
-        }
+    // handle a function's source parameters
+    for (const auto &Idx : FunDesc.SourceValues) {
+      if (Idx >= Fun->arg_size()) {
+        llvm::errs() << "ERROR: The source-function parameter index is out of "
+                        "bounds: "
+                     << Idx << "\n";
+        // Use 'continue' instead of 'break' to get error messages for the
+        // remaining parameters as well
+        continue;
       }
-      if (Params.contains("sink")) {
-        for (const auto &Idx : Params["sink"]) {
-          if (Idx.is_number()) {
-            if (Idx >= Fun->arg_size()) {
-              llvm::errs()
-                  << "ERROR: The source-function parameter index is out of "
-                     "bounds: "
-                  << Idx << "\n";
-              continue;
-            }
-            addTaintCategory(Fun->getArg(Idx), TaintCategory::Sink);
-          } else if (Idx.is_string()) {
-            const auto Sinks = Idx.get<std::string>();
-            if (Sinks == "all") {
-              for (const auto &Arg : Fun->args()) {
-                addTaintCategory(&Arg, TaintCategory::Sink);
-              }
-            }
-          }
-        }
+
+      addTaintCategory(Fun->getArg(Idx), TaintCategory::Source);
+    }
+    for (const auto &Idx : FunDesc.SinkValues) {
+      if (Idx >= Fun->arg_size()) {
+        llvm::errs() << "ERROR: The sink-function parameter index is out of "
+                        "bounds: "
+                     << Idx << "\n";
+        continue;
       }
-      if (Params.contains("sanitizer")) {
-        for (unsigned Idx : Params["sanitizer"]) {
-          if (Idx >= Fun->arg_size()) {
-            llvm::errs()
-                << "ERROR: The source-function parameter index is out of "
-                   "bounds: "
-                << Idx << "\n";
-            continue;
-          }
-          addTaintCategory(Fun->getArg(Idx), TaintCategory::Sanitizer);
+      addTaintCategory(Fun->getArg(Idx), TaintCategory::Sink);
+    }
+
+    for (const auto &Idx : FunDesc.SinkStringValues) {
+      if (Idx == "all") {
```

Comment:
Better have a flag in the FunctionData that tells whether we have an "all" sink-param